src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs

```diff
@@ -21,15 +21,31 @@ public static class TextLoaderSaverCatalog
         /// <param name="catalog">The <see cref="DataOperationsCatalog"/> catalog.</param>
         /// <param name="columns">Array of columns <see cref="TextLoader.Column"/> defining the schema.</param>
         /// <param name="separatorChar">The character used as separator between data points in a row. By default the tab character is used as separator.</param>
-        /// <param name="hasHeader">Whether the file has a header with feature names. Note: If a TextLoader is created with hasHeader = true but without a
-        /// <paramref name="dataSample"/>, then vector columns made by TextLoader will not contain slot name annotations (slots being the elements of the given vector column),
-        /// because the output schema is made when the TextLoader is made, and not when <see cref="TextLoader.Load(IMultiStreamSource)"/> is called.
-        /// In addition, the case where dataSample = null and hasHeader = true indicates to the loader that when it is given a file when Load()
-        /// is called, it needs to skip the first line.</param>
-        /// <param name="dataSample">The optional location of a data sample. The sample can be used to infer column names and number of slots in each column.</param>
-        /// <param name="allowQuoting">Whether the file can contain columns defined by a quoted string.</param>
-        /// <param name="trimWhitespace">Remove trailing whitespace from lines</param>
-        /// <param name="allowSparse">Whether the file can contain numerical vectors in sparse format.</param>
+        /// <param name="hasHeader">Whether the file has a header with feature names. When <see langword="true"/>, the loader will skip the first line when
+        /// <see cref="TextLoader.Load(IMultiStreamSource)"/> is called. In addition, when <paramref name="hasHeader"/> is <see langword="true"/> but
+        /// <paramref name="dataSample"/> is not provided, vector columns created by the loader will not contain slot name annotations (slots being the elements of the given vector column),
+        /// because the output schema is made when the loader is created, and not when <see cref="TextLoader.Load(IMultiStreamSource)"/> is called.</param>
+        /// <param name="dataSample">The optional location of a data sample. The sample can be used to infer slot name annotations if present, and also the number
+        /// of slots in a column defined with <see cref="TextLoader.Range"/> with <see langword="null"/> maximum index.
+        /// If the sample has been saved with ML.NET's <see cref="SaveAsText(DataOperationsCatalog, IDataView, Stream, char, bool, bool, bool, bool)"/>,
+        /// it will also contain the schema information in the header that the loader can read even if <paramref name="columns"/> is <see langword="null"/>.</param>
```

Comment:
>even if <paramref name="columns"/> is <see langword="null"/> [](start = 95, length = 60)

In order to use the schema defined in the file, all other arguments should be left with their default values (including null for the columns).